### PR TITLE
Extended planning service with validator

### PR DESCRIPTION
**Pull request by likelion**

### CODE REVIEW

**Comment by likelion** [comment]
Example of capturing output of the validator by using `pddl:adding`, `pddl:deleting`, and `pddl:updating` properties in the existing plan (`pddl:current` is used to capture the current value of a fluent before the update is applied as returned by the validator):

```turtle
<plan>
        a pddl:Plan ;
        pddl:cost 2.0E0 ;
        pddl:step [ a pddl:Step ;
                    pddl:action [ a :move ;
                                  :move-b :c ;
                                  :move-x :a ;
                                  :move-y :table
                                ] ;
                    pddl:adding [ a :on ;
                                  :on-x :move-b ;
                                  :on-y :move-y
                                ] ;
                    pddl:deleting [ a :on ;
                                    :on-x :move-b ;
                                    :on-y :move-x
                                  ] ;
                    pddl:updating [ a pddl:Increase ;
                                    pddl:current 0 ;
                                    pddl:parameter [ a :moved ;
                                                     :moved-m :move-b
                                                   ] ;
                                    pddl:argument 1
                                  ] ;
                    oslc:instanceShape pddl:StepShape ;
                    sh:order 1
                  ] ,
                  [ a pddl:Step ;
                    pddl:action [ a :move ;
                                  :move-b :a ;
                                  :move-x :table ;
                                  :move-y :c
                                ] ;
                    oslc:instanceShape pddl:StepShape ;
                    sh:order 2
                  ] ;
        pddl:time 0.0E0 ;
        oslc:instanceShape pddl:PlanShape .
```

**Comment by likelion** [comment]
Predicate `pddl_generator:generate_pddl/3` now supports resources of type `pddl:Plan` and generates PDDL text compatible with input of VAL validator.

**Comment by swarupmohalik** [comment]
@likelion , should **Imply** also have left and right, since it is binary and the order is important?
How is it currently handled?

e.g. 
--b [ a pddl:Imply ;
        pddl:argument
                       [a pddl:And ;
                       ..
                       ] ,
                       [a pddl:Or ;
                       ..
                       ]
         ]

**Comment by likelion** [comment]
@swarupmohalik Yes, `pddl:Imply` must use `pddl:left` and `pddl:right` because the order is important.

File: oslc_prolog/lib/oslc_shape.pl [hunk 1]
```diff
@@ -51,14 +51,16 @@
...
-  once(rdf(PropertyResource, oslc:valueType, Type)),
-  once((
-    check_value(IRI, PropertyResource, V, Type),
-    check_allowed_values(IRI, PropertyResource, V, Type)
-  ; rdf(PropertyResource, oslc:propertyDefinition, PropertyDefinition),
-    oslc_error("Property [~w] of resource [~w] must be of type [~w]", [PropertyDefinition, IRI, Type])
-  )),
-  check_value_type(IRI, PropertyResource, T, Type).
+  (once(rdf(PropertyResource, oslc:valueType, Type))
+  -> once((
```
Comment (likelion) [inline oslc_prolog/lib/oslc_shape.pl:55]: FIX: if the `oslc:valueType` property is omitted, the shape should not do any type checking.

**Review by kicacacko — APPROVED** [review]

**Pull request merged**